src/plugins/intel_cpu/tests/unit/snippets_transformations/lowered/buffer_allocation.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/opsets/opset.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/pass/mark_loops.hpp"
+#include "snippets/lowered/pass/init_loops.hpp"
+#include "snippets/lowered/pass/insert_load_store.hpp"
+#include "snippets/lowered/pass/validate_loops.hpp"
+#include "snippets/lowered/pass/insert_loops.hpp"
+#include "snippets/lowered/pass/allocate_buffers.hpp"
+#include "snippets/lowered/pass/fuse_loops.hpp"
+#include "snippets/lowered/pass/split_loops.hpp"
+#include "snippets/lowered/pass/insert_buffers.hpp"
+#include "snippets/lowered/pass/softmax_decomposition.hpp"
+
+#include "transformations/snippets/x64/shape_inference.hpp"
+#include "transformations/snippets/x64/pass/lowered/brgemm_blocking.hpp"
+#include "transformations/snippets/x64/pass/lowered/set_brgemm_copy_b_buffers_shape.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "transformations/snippets/x64/op/brgemm_copy_b.hpp"
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "common_test_utils/common_utils.hpp"
+
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+/*  Note[74841]:
```

Comment:
Does it make sense to create this test at all than? Does it provide any additional coverage?

Reply:
These tests are needed to validate Buffer Allocation algorithm on Linear IRs after CPU specific transformations. For example, CPU Plugin has transformation that convert `Brgemm` to subgraph of `Brgemm`, `BrgemmCopyB` and `Buffers`  ops in I8, BF16 cases. Moreover, for AMX case CPU adds `Buffer` with new memory for `Brgemm`. We have `Buffer` with new memory only there.

Also I found the bug with this Buffer on SPR that the memory was allocated not optimally. After fix I decided to add this test `MHABF16AMXBufferAllocationTest` to validate this case.